### PR TITLE
feat(convex): add investment limit

### PR DESCRIPTION
**Pull request by dhruvinparikh**
Fixes #Convex Finance Investment limit

#### PR Checklist

- [X] Tests
- [X] Bump up Solidity version for Convex Adapter

### CODE REVIEW

**Review by leodinh — DISMISSED** [review]
Should we increase the solidity version to remove SafeMath?

File: contracts/utils/AdapterInvestLimitBase.sol [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+        require(_liquidityPool.isContract(), "!_liquidityPool.isContract()");
+        require(_underlyingToken.isContract(), "!_underlyingToken.isContract()");
+        maxDepositAmount[_liquidityPool][_underlyingToken] = _maxDepositAmount;
+        emit LogMaxDepositAmount(maxDepositAmount[_liquidityPool][_underlyingToken], msg.sender);
+    }
+
+    /**
+     * @inheritdoc IAdapterInvestLimit
+     */
+    function setMaxDepositProtocolMode(MaxExposure _mode) public override onlyRiskOperator {
```
Comment (deepcrazy) [inline contracts/utils/AdapterInvestLimitBase.sol]:
```suggestion
    function setMaxDepositProtocolMode(MaxExposure _mode) external override onlyRiskOperator {
```

File: contracts/utils/AdapterInvestLimitBase.sol [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+        maxDepositProtocolMode = _mode;
+        emit LogMaxDepositProtocolMode(maxDepositProtocolMode, msg.sender);
+    }
+
+    /**
+     * @inheritdoc IAdapterInvestLimit
+     */
+    function setMaxDepositProtocolPct(uint256 _maxDepositProtocolPct) public override onlyRiskOperator {
```
Comment (deepcrazy) [inline contracts/utils/AdapterInvestLimitBase.sol]:
```suggestion
    function setMaxDepositProtocolPct(uint256 _maxDepositProtocolPct) external override onlyRiskOperator {
```

**Review by deepcrazy — CHANGES_REQUESTED** [review]
Request for 1 minor change in access modifiers in AdapterInvestLimitBase.sol.

**Review by deepcrazy — APPROVED** [review]

**Review by leodinh — APPROVED** [review]

**Pull request merged**